client/client.go

```diff
@@ -193,21 +207,27 @@ func (c *client) tsCancelLoop() {
 	ctx, cancel := context.WithCancel(c.ctx)
 	defer cancel()
 
-	for {
-		select {
-		case d := <-c.tsDeadlineCh:
-			select {
-			case <-d.timer:
-				log.Error("tso request is canceled due to timeout", errs.ZapError(errs.ErrClientGetTSOTimeout))
-				d.cancel()
-			case <-d.done:
-			case <-ctx.Done():
-				return
+	// Watch every dc-location's tsDeadlineCh
+	for dcLocation, deadlineCh := range c.tsDeadlineCh {
```

Comment:
What if new DC added after client started?

Reply:
Nice catch. I will update it later.